File: src/my_project/easy_problems/from151To200/find_middle_index.py
from typing import List 

class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:

        len_nums: int = len(nums)
        for i in range(len(nums)):
            if sum(nums[0:i]) == sum(nums[i+1:len_nums]):
                return i
        if len_nums == 2:
            return -1
        if len_nums == 1:
            return 0

        return -1

solution = Solution()

print(solution.findMiddleIndex(nums = [2,3,-1,8,4]))

print(solution.findMiddleIndex(nums = [1,-1,4]))

print(solution.findMiddleIndex(nums = [2,5]))

print(solution.findMiddleIndex(nums = [1]))


        

File: src/my_project_typescript/easy_problems_typescript/from51To100/findMiddleIndex.ts
function findMiddleIndex(nums: number[]): number {

    let lenNums: number = nums.length
    for (let i = 0; i < lenNums; i++){
        if (nums.slice(0,i).reduce((a,b) => a+b,0) === nums.slice(i+1).reduce((a,b) => a+b,0)){
            return i
        }
    }

    if (lenNums === 2){
        return -1
    }
    if (lenNums ===1){
        return 0
    }

    return -1
};

console.log(findMiddleIndex([2,3,-1,8,4]))

console.log(findMiddleIndex([1,-1,4]))

console.log(findMiddleIndex([2,5]))

console.log(findMiddleIndex( [1]))
